### ISSUE TITLE
MISSING JWT FIELD DOCUMENTATION

### ISSUE THREAD

**Issue by eyyyyyyy3** [supabase/supabase#26720]
# Improve documentation
Add a documentation page for the JWT fields.

## Describe the problem
When trying to write a server the JWT needs to be serialized. For that, the programmer needs to know the fields, their types and whether they are optional or not.

## Additional context
I am authenticating the JWT in Rust and I need to know all the fields and their necessities. Additionally fields like `ref` are reserved keywords in Rust and they might pose a problem.

**Comment by swamidass** [comment]
I agree this is an issue. There is more to validation than this, it seems. This may be related: https://github.com/supabase/supabase/issues/26716

**Comment by swamidass** [comment]
Turns out almost everything is optional. All you need is "role". Obviously, that wouldn't be best practices. You should at least include an iat and exp too. See the solution here: https://github.com/supabase/supabase/issues/26716

**Comment by eyyyyyyy3** [comment]
Cmon guys. Kinda need this ngl.

**Comment by j4w8n** [comment]
This is what I came up with for my own purposes.

https://github.com/j4w8n/sveltekit-supabase-ssr/blob/main/src%2Ftypes.ts

**Comment by Surbhideep03** [comment]
;;;;

**Comment by Surbhideep03** [comment]
I have pulled a request about it please check it

**Comment by swamidass** [comment]
which pr?

**Comment by Surbhideep03** [comment]
the missing JWT Field Documentation

**Comment by Hallidayo** [comment]
Hi @Surbhideep03, thank you for the PR have you a link?

**Comment by Vangmay** [comment]
Hello, I worked on this issue. Here is my PR https://github.com/supabase/supabase/pull/36578
